components/mailrelay/keyvault.tf

```diff
@@ -30,8 +30,8 @@ resource "azurerm_role_assignment" "acme" {
 }
 
 locals {
-  # Needed for role assignment only
   wi_environment = var.env == "dev" ? "stg" : var.env
+  product_list   = var.env == "prod" ? toset(["mailrelay2"]) : toset(["mailrelay", "mailrelay2"])
```

Comment:
Would you mind adding code comment about why this has to be done this way so it makes sense in the future?